app/document/input/mouse.js

```diff
@@ -132,47 +132,43 @@ class MouseListener extends events.EventEmitter {
         if (event.ctrlKey) { // zooming
             event.preventDefault();
             if (this.listening_to_wheel) {
-                if (event.deltaY > 5) {
+                if (event.deltaY > 0) {
                     zoom_out();
-                } else if (event.deltaY < 5) {
+                } else if (event.deltaY < 0) {
                     zoom_in();
                 }
                 this.listening_to_wheel = false;
                 setTimeout(() => {
                     this.listening_to_wheel = true;
-                }, 50);
+                }, 100);
             }
         } else if (event.shiftKey) { // reference image opacity
             if (this.listening_to_wheel) {
-                let e = document.getElementById("reference_image");
-                let o = parseFloat(e.style.opacity);
-                let a = 0.2;
-                if (event.deltaY > 5) {
-                    if (o >= a) o = o - a;
-                } else if (event.deltaY < 5) {
-                    if (o <= (1.0 - a)) o = o + a;
+                if (event.deltaX > 0 || event.deltaY > 0) {
```

Comment:
This is one part where I didn't understand the existing code, or why the magic number "5" was chosen for the cutoff in positive/negative scrolling?

At least on my system (macbook air m1 + razer deathaddr v2 mouse):

- Scrolling down is registered as `deltaX = 4.0`
- Scrolling up is registered as `deltaX = -4.0`
- Holding down the shift key changes the delta on scrolling from `deltaX` to `deltaY`
- Using the touchpad to pinch-to-zoom gives me either positive or negative values centered around `0`.

So scrolling was pretty much broken & glitchy for me without making these changes (I could zoom in but I couldn't zoom out).

Reply:
I'm not sure on the magic number either. maybe a resolution thing? some kind of threshold boundary that is crossed? maybe it was too sensitive? hrm.